lib/setup_tools.rb

```diff
@@ -74,7 +74,29 @@ def set_value(key, value, options = {})
   def ask(question, opts = {})
     print question + " "
     STDOUT.flush
-    (opts[:noecho] ? STDIN.noecho(&:gets) : gets).strip
+    if opts[:noecho] && is_noecho_unsupported
+      ask_without_echo.strip
+    else
+      (opts[:noecho] ? STDIN.noecho(&:gets) : gets).strip
+    end
+  end
+
+  def is_noecho_unsupported
+    return RbConfig::CONFIG['host_os'] =~ /cygwin|mswin|mingw32/
```

Comment:
This condition is wrong.
Only msys mintty is the case.

Reply:
When I run Ruby installed by RubyInstaller 64bit (x64-mingw32) on MinGW-w64 Win64 Shell I see this happen:

```
knu@Gryphon% /c/Ruby/2.6/bin/ruby -v -rrbconfig -rio/console -e "p RbConfig::CONFIG['host_os']; p STDIN.noecho(&:gets)"
ruby 2.6.6p146 (2020-03-31 revision 67876) [x64-mingw32]
"mingw32"
-e:1:in `noecho': Bad file descriptor (Errno::EBADF)
        from -e:1:in `<main>'
```